src/app/game.ts

```diff
@@ -430,6 +438,19 @@ export class Game {
     return this.buildings.find((building) => building instanceof Wall) as Wall
   }
 
+  private getBarracks(): Barracks | undefined {
+    return this.buildings.find(
+      (building) => building instanceof Barracks
+    ) as Barracks
+  }
+
+  getBarracksLevel(): number {
```

Comment:
I think it would be better to game not have such function, you can use getBarracks() function and then get level from Barrack object.